doc/ExpertOptions.md

```diff
@@ -1555,3 +1556,66 @@ dx describe /builds/1.02/applets/hello --json --details | jq '.details | .wdlSou
 # Recompilation
 
 Any significant WDL workflow is compiled into multiple DNAnexus applets and workflows. Naively, any modification to the WDL source would necessitate recompilation of all the constituent objects, which is expensive. To optimize this use case, all generated platform objects are checksumed. If a dx:object has not changed, it is not recompiled, and the existing version can be used. The checksum covers the WDL source code, the DNAnexus runtime specification, and any other attributes. There are two exceptions: the project name, and the folder. This allows moving WDL workflows in the folder hierarchy without recompilation.
+
+# Publishing global workflows
+
+<!-- TODO supported version pending release w/ APPS-989 -->
+Publishing a dxCompiler WDL workflow as a global workflow is supported from dxCompiler >= `v2.8.0` and dxpy >= `v<supported version here>`
+
+Example: compiling a WDL workflow for later use as a global workflow.
+```
+java -jar dxCompiler.jar compile <workflow name>.wdl -instanceTypeSelection dynamic
+```
+
+<!-- TODO edit example to use <project id>:<workflow id> once dxpy supports that -->
+Example: publishing a global workflow from a compiled workflow. The global workflow's name will match the WDL workflow name. The global workflow's version must be set with `--version`, since a non-global workflow has no such property. If `--bill-to` is not specified, the publishing user's default billTo will be assumed.
+```
+dx build --globalworkflow --from <workflow id> --version <version> --bill-to <user-xxxx | org-yyyy>
+dx publish globalworkflow-<workflow name>/<version>
+```
+
+Example: adding authorized users to a global workflow
+```
+dx add users globalworkflow-<workflow name> <user-xxxx | org-yyyy | PUBLIC>
+```
+
+See [Limitations](#limitations) below for more details on which parts of the workflow will be automatically included in the global workflow.
+
+## Recommendations
+
+Developers should not store any credentials (passwords, keys, etc.) in the source code of the global workflow, as authorized users will have permission to download and view any applets referenced in the global workflow.
+
+Developers should use simple types for input/output interfaces for global workflows for better platform experience.
+
+For better portability across projects where the workflow will be run, developers should specify runtime resources using memory, disk, CPU requirements (not hard-coded instance types using the key `dx_instance_type`). Developers should compile WDL workflows with `-instanceTypeSelection dynamic`. Expect slightly longer runtimes due to the need to pick instance types at runtime, but this ensures the workflow will always use instance types that are supported for the project where it runs.
+
+For informational purposes, developers should include a reference to a git repo commit containing the original source code. They can add this to the "description" metadata field of the global workflow.
```

Comment:
How about adding a link to the documentation where we explain metadata so the users can quickly see how to do it? Or adding an example/snippet here.

Reply:
To do: Add examples of how to update the WF metadata (title, summary, description, tags, categories) by API call after the global WF is published.

Note: it looks like it will have to be done that way, see discussion on ticket linked to testing ticket APPS-627.